docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,485 @@
+will claim the device, by setting `metadata.owner` to `Fleet/{fleet-name}`.
+
+The `os.image` and configuration fields in this API support floating tags, and while the fleet manager
+does freeze the tags to hashes, the user is free to setup the device spec with floating references.
+
+###### Expected behaviour regarding external references with floating tags
+
+Changes on external references is monitored and acted on (git branches,
+device images, secrets), the device controller should re-evaluate the device
+spec and apply the changes to the rendered version which will be exposed
+on the `/device/{name}/rendered` endpoint.
+
+Several mechanisms can be used to track external references:
+* Polling
+* Webhooks
+
+##### `/device/{name}` while managed by a fleet (Fleet Device)
+
+When managing a device through a fleet, flightctl is responsible
+for maintaining the device spec in sync with the fleet spec; in this case
+*write access is only possible for the modification of labels and other metadata*
+through this API except for the owner fleet.
+
+> [!NOTE]
+> Devices managed by a fleet have their floating tags resolved to stable hashes at this API
+> level. This is done to ensure that the device is always in sync with the fleet spec.
+
+The device will have an owner reference to the managing fleet, i.e.:
+
+```yaml
+  owner: Fleet/optical-inspector-production-fleet
+```
+
+###### Temporarily disabling fleet management (Fleet Device paused)
+
+A user can temporarily stop fleet management of a device by setting the `flightctl.io/fleet-controller=paused`
+in the device labels, the fleet manager will ignore devices with this label for any subsequent
+update or rollout.
+
+This feature is intended to allow temporarily debugging a device without de-associating
+from the fleet. This will mean the the `metadata.owner` field is not cleared. 
+
+Devices set to `flightctl.io/fleet-controller=paused` with a `metadata.owner` fleet must signal
+a device status condition to indicate that the device is not being synchronized
+to the owner fleet.
+
+```yaml
+  - lastTransitionTime: "2024-04-30T15:06:20Z"
+    message: This device is not in sync with the owner fleet.
+    reason: Management Paused from flightctl.io/fleet-controller=paused label.
+    status: False
+    type: Management
+```
+
+When a fleet owned device is moved back to fleet management (the label is cleared),
+the status condition is removed, in additiona the fleet controller will roll
+out (or back) the device configuration to the fleet mandated configuration and remove any
+modifications manually made to the device spec.
+
+###### Removing a device from a fleet
+
+A device could be stripped from the labels associating it to a specific fleet, in such
+case the `metadata.owner` field will be cleared, and the device will need to be manually managed
+or eventually associated to a different fleet.
+
+in this case a condition will be set to indicate that the device is not managed by any fleet.
```

Comment:
This condition would be inconsistent with devices that were never managed. I suggest we remove it. it should be clear by the empty owner field.

Reply:
I agree. It mirrors the former management mode "manual" vs "fleet" spec item.

Reply:
Done.